src/librustc_trans/debuginfo/create_scope_map.rs

```diff
@@ -113,7 +133,12 @@ fn make_mir_scope(ccx: &CrateContext,
         scopes[parent]
     } else {
         // The root is the function itself.
-        scopes[scope] = fn_metadata;
+        let loc = span_start(ccx, mir.span);
+        scopes[scope] = MirDebugScope {
+            scope_metadata: fn_metadata,
+            start_pos: loc.file.start_pos,
+            end_pos: loc.file.end_pos,
```

Comment:
Wouldn't it be more correct to have the start and end pos of the function instead of the file containing it?